src/raw/cgroupv1.go

```diff
@@ -38,20 +49,18 @@ func (cg *CgroupsV1Fetcher) Fetch(c types.ContainerJSON) (Metrics, error) {
 	pid := c.State.Pid
 	containerID := c.ID
 
-	var (
-		cgroupInfo *cgroupV1Paths
-		err        error
-	)
-	cgroupInfo, err = getCgroupV1Paths(cg.hostRoot, pid)
-
+	err := cg.cgroupDetector.PopulatePaths(cg.hostRoot, pid)
 	if err != nil {
 		return stats, err
 	}
 
+	cgroupInfo := cg.cgroupDetector.(*CgoupsV1Detector).paths
```

Comment:
Maybe I'd feel more confortable if we check for the value in the type assertion, wdyt?

Reply:
This is now changed and it's not needed anymore.

Reply:
The integrations are now separated per version